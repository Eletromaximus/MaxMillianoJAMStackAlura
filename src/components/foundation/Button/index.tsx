import React from 'react'
import styled from 'styled-components'
import propToStyle from '../../../theme/utils/propToStyle'
import { TextStyleVariants } from '../Text'
import Link from '../Link'
// import get from 'lodash/get'

interface IButton {
  variant: string;
  target: string;
  children: React.ReactNode;
  onClick?: (e?: Event) => void;
  href?: string;
  paddingRight?: string | object;
  padding?: string | object | number;
  margin?: string | object | number;
  name?: string;
  color?: string;
  width?: string;
}

const ButtonBase = styled.button<IButton>`
  background-color: transparent;
  border: none;
  
  ${(props) => TextStyleVariants[props.variant]};
  ${propToStyle('color')};
  ${propToStyle('paddingRight')};
  ${propToStyle('margin')};
  ${propToStyle('padding')};
  ${propToStyle('width')};
`
// color: ${({ theme, variant }) => get(theme, `colors.${variant}.color`)}
export default function Button ({ variant, href, children, ...props }: IButton) {
  const hasTag = Boolean(href)
  const tag = hasTag ? Link : 'button'

  return (
    <ButtonBase as={tag} variant={variant} href={href} {...props} >
      {children}
    </ButtonBase>
  )
}

Button.defaultProps = {
  variant: 'paragraph1',
  target: '_blanck',
  href: undefined
}
