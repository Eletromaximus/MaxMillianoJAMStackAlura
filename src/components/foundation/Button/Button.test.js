import React from 'react'
import Button from './index'
import renderer from 'react-test-renderer'

describe('button style with and without href', () => {
  test('it applies default styles button without href', () => {
    const tree = renderer.create(<Button variant='paragraph1' />).toJSON()

    expect(tree).toMatchSnapshot()
  // expect(tree).toHaveStyleRule('font-size', '1.75rem')
  // expect(tree).toHaveStyleRule('font-weight', '400')
  // expect(tree).toHaveStyleRule(
  //   'font-family',
  //   'Fira Sans Condensed,sans-serif'
  // )
  })
  // eslint-disable-next-line jest/no-commented-out-tests
  // test('it applies default styles button with href', () => {
  //   const tree = renderer.create(<Button variant='paragraph1' href='/' />).toJSON()

  //   expect(tree).toMatchSnapshot()
  // // expect(tree).toHaveStyleRule('font-size', '1.75rem')
  // // expect(tree).toHaveStyleRule('font-weight', '400')
  // // expect(tree).toHaveStyleRule(
  // //   'font-family',
  // //   'Fira Sans Condensed,sans-serif'
  // // )
  // })
})
