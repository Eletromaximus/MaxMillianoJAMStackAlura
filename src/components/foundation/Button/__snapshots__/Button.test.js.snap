// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`button style with and without href it applies default styles button without href 1`] = `
.c0 {
  background-color: transparent;
  border: none;
  font-size: 16px;
  font-weight: 400;
  line-height: 1.25;
}

<button
  className="c0"
/>
`;

exports[`it applies default styles 1`] = `
.c0 {
  background-color: transparent;
  border: none;
}

<button
  className="c0"
/>
`;

exports[`it applies default styles without href 1`] = `
.c0 {
  background-color: transparent;
  border: none;
  font-size: 16px;
  font-weight: 400;
  line-height: 1.25;
}

<button
  className="c0"
/>
`;
